src/datalab_app_plugin_insitu/nmr_insitu.py

```diff
@@ -8,6 +8,44 @@
 from .utils import _process_data
 
 
+def _find_folder_path(base_path: Path, target_folder_name: str) -> Optional[Path]:
+    """
+    Find a folder path recursively regardless of nesting level.
```

Comment:
I get what you mean, but is this a better explanation? (I'm not sure)

```suggestion
    Find a folder path inside a zip regardless of whether the zip has an extra level of nesting.
```